File: src/main/java/com/dochiri/hexagonal/presentation/auth/response/UserProfileResponse.java
package com.dochiri.hexagonal.presentation.auth.response;

import com.dochiri.hexagonal.application.user.dto.UserProfileResult;

public record UserProfileResponse(
        String userId,
        String email,
        String name,
        String role
) {
    public static UserProfileResponse from(UserProfileResult userProfileResult) {
        return new UserProfileResponse(
                userProfileResult.userId(),
                userProfileResult.email(),
                userProfileResult.name(),
                userProfileResult.role()
        );
    }
}

File: src/main/kotlin/com/dochiri/hexagonal/presentation/auth/response/UserProfileResponse.kt
package com.dochiri.hexagonal.presentation.auth.response

import com.dochiri.hexagonal.application.user.dto.UserProfileResult

data class UserProfileResponse(
    val userId: String,
    val email: String,
    val name: String,
    val role: String
) {
    companion object {
        fun from(userProfileResult: UserProfileResult): UserProfileResponse =
            UserProfileResponse(
                userProfileResult.userId,
                userProfileResult.email,
                userProfileResult.name,
                userProfileResult.role
            )
    }
}
